packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -267,6 +272,19 @@ export class AccountTreeController extends BaseController<
     this.update((state) => {
       state.accountTree.wallets = wallets;
 
+      // Apply group metadata within the state update
+      for (const [walletId, wallet] of Object.entries(
+        state.accountTree.wallets,
+      )) {
+        for (const groupId of Object.keys(wallet.groups)) {
+          this.#applyAccountGroupMetadata(
+            state,
+            walletId as AccountWalletId,
+            groupId as AccountGroupId,
+          );
+        }
+      }
```

Comment:
We can probably re-use the original pattern, which will avoid those type-casts

```suggestion
      for (const wallet of Object.values(wallets)) {
        for (const group of Object.values(wallet.groups)) {
          this.#applyAccountGroupMetadata(state, wallet.id, group.id);
        }
      }
```